add info about post-processing query (#997)
neo4j/docs-aura: modules/ROOT/pages/aura-agent.adoc

Reviewer: ```suggestion

[NOTE]
====
Use the *Similarity Search* tool when your starting point is a vector index.    
Use the *Cypher Template* tool for arbitrary read queries that do not require a
vector lookup.
====

```

This suggestion is to replace the `Keep in mind that adding this post-processing query to the similarity search narrows the scope of the tool.` part above.

Reviewer: ```suggestion
This tool uses vector similarity to do a semantic search.
```

The suggested note below covers the "complement" relationship more clearly. Also, with this new addition, the vector search tool covers the cypher template use case.

You need to provide the name of your vector index and a value for *Top K*, i.e. the number of results you want the search to return.

If you want to further refine the tool, you can use the results from the similarity search (`node` and `score`) in a Cypher query.
When you check the box *Return connected graph data*, you have two options.
Either use natural language to describe what you want the query to do  and then use *Generate* to get a corresponding Cypher query.
Review the query and make any necessary changes, the use *Add tool* to confirm.
The other option is to write your own Cypher query from the *Cypher* tab, for the purpose.

Keep in mind that adding this post-processing query to the similarity search narrows the scope of the tool.

> But it is also possible to have multiple similarity search tools with the same vector search configuration, but with different post-processing queries.

Reviewer: ```suggestion
To further process the similarity search results, check *Return connected graph 
data*.
This lets you write a Cypher query that is appended to the vector search and
runs as a single query, with `node` (the matched node) and `score` (the         
similarity score) already in scope.

You can write the query directly on the *Cypher* tab, or describe the traversal 
on the *Natural language* tab and use *Generate* to produce a Cypher query.
Review and edit the query on the *Cypher* tab, then use *Add tool* to save it.  
Note that the natural language description is not saved — only the final Cypher
query is stored with the tool.

You can add multiple similarity search tools with the same vector index but     
different post-processing queries to cover different traversal patterns.
```

My suggestion replaces the generic "refine the tool" wording with "further process the results" and highlights the fact that the user has to provide a Cypher query (which can be added manually or generated), and avoid any possible confusion around the natural language query.
